builder/vmware/iso/builder.go

```diff
@@ -120,6 +120,10 @@ func (b *Builder) Run(ctx context.Context, ui packersdk.Ui, hook packersdk.Hook)
 		&vmwcommon.StepSuppressMessages{},
 		&vmwcommon.StepHTTPIPDiscover{},
 		commonsteps.HTTPServerFromHTTPConfig(&b.config.HTTPConfig),
+		&communicator.StepSSHKeyGen{
```

Comment:
We should only call StepSSHKeyGen if the configured communicator is "ssh" otherwise a SSH key will get generated when using winrm or no communicator at all. Which might be confusing. 

```suggestion
		multistep.If(b.config.Comm.Type == "ssh", &communicator.StepSSHKeyGen{
```